struct.md

```diff
@@ -0,0 +1,297 @@
+# Struct Proposal
+
+This document tries to define with examples what shall be (theoretically) supported in VSS after adding struct support.
+I.e. what you can do still claiming that the model is correct VSS.
+It only to a limited extent show implications for vss-tools, and then only for syntactic/semantic checks.
+As of now it does not state how exporters are affected.
+
+## Proposed VSS 4.0 acceptance criteria and increments
+
+We need to decide how far we must go before we can release VSS 4.0, we does not need to go the whole way in one step
+
+### Increment 1
+
+* Syntax can be used without VSS-tools complaining
+* No semantic check by VSS-tools
+* No documentation
+* Not used in VSS standard catalog
+* Struct not supported by VSS-tools exporters
+
+
+### Increment 2
+
+* Semantic check on type-references by VSS-tools
+* Syntax documented
+* Well defined behavior for all exporters (e.g. either support, and/or give warning if used)
+
+### Increment 3
+
+* All "standard" exporters in vss-tools support structs
+* Guidelines on when to consider using structs rather than branches documented
+
+### Increment 4
+
+* We may start to use structs in VSS standard catalog.
+
+
+## Simple Usage
+
+```
+DeliveryInfo:
+  type: struct
+  description: A struct type containing info for each delivery
+  
+DeliveryInfo.Address:
+  datatype: string
+  type: item
+  description: Destination address
+
+DeliveryInfo.Receiver:
+  datatype: string
+  type: item
+  description: Name of receiver
+
+DeliveryList:
+  datatype: DeliveryInfo
+  type: sensor
+  description: List of deliveries
+```
+
+For VSS 4.0 it is not necessary that vss-tools do semantic check, i.e. if someone would add an extra `f` by mistake like this:
+
+```
+
+DeliveryList:
+  datatype: DeliveryInffo
```

Comment:
Spelling *Inffo*

Reply:
It is deliberate, see line 60

Reply:
Ah yes, ok understood. I spotted it as I passed and should have read the surrounding text. We can consider this comment closed.